geomesa-convert/README.md~

```diff
@@ -0,0 +1,785 @@
+
+Example: ```stripQuotes('fo"o') = foo```
+
+#### length
+
+Description: Returns the length of a string.
+
+Usage: ```length($1)```
+
+Example: ```length('foo') = 3```
+
+#### trim
+
+Description: Trim whitespace from around a string.
+
+Usage: ```trim($1)```
+
+Example: ```trim('  foo ') = foo```
+
+#### capitalize
+
+Description: Capitalize a string.
+
+Usage: ```capitalize($1)```
+
+Example: ```capitalize('foo') = Foo```
+
+#### lowercase
+
+Description: Lowercase a string.
+
+Usage: ```lowercase($1)```
+
+Example: ```lowercase('FOO') = foo```
+
+#### uppercase
+
+Description: Uppercase a string.
+
+Usage: ```uppercase($1)```
+
+Example: ```uppercase('foo') = FOO```
+
+#### regexReplace
+
+Description: Replace a given pattern with a target pattern in a string.
+
+Usage: ```regexReplace($regex, $replacement, $1)```
+
+Example: ```regexReplace('foo'::r, 'bar', 'foobar') = barbar```
+
+#### concatenate
+
+Description: Concatenate two strings.
+
+Usage: ```concatenate($0, $1)```
+
+Example: ```concatenate('foo', 'bar') = foobar```
+
+#### substring
+
+Description: Return the substring of a string.
+
+Usage: ```substring($1, $startIndex, $endIndex)```
+
+Example: ```substring('foobarbaz', 2, 5) = oba```
+
+#### toString
+
+Description: Convert another data type to a string.
+
+Usage: ```toString($0)```
+
+Example: ```concatenate(toString(5), toString(6)) = '56'```
+ 
+### Date Functions
+
+#### now
+
+Description: Use the current system time.
+
+Usage: ```now()```
+
+#### date
+
+Description: Custom date parser.
+
+Usage: ```date($format, $1)```
+
+Example: ```date('YYYY-MM-dd\'T\'HH:mm:ss.SSSSSS', '2015-01-01T00:00:00.000000')```
+
+#### dateTime
+
+Description: A strict ISO 8601 Date parser for format ```yyyy-MM-dd'T'HH:mm:ss.SSSZZ```.
+ 
+Usage: ```dateTime($1)```
+
+Example: ```dateTime('2015-01-01T00:00:00.000Z')```
+
+#### basicDate
+
+Description: A basic date format for ```yyyyMMdd```.
+
+Usage: ```basicDate($1)```
+
+Example: ```basicDate('20150101')```
+
+#### basicDateTime
+
+Description: A basic format that combines a basic date and time for format ```yyyyMMdd'T'HHmmss.SSSZ```.
+
+Usage: ```basicDateTime($1)```
+
+Example: ```basicDateTime('20150101T000000.000Z')```
+
+#### basicDateTimeNoMillis
+
+Description: A basic format that combines a basic date and time with no millis for format ```yyyyMMdd'T'HHmmssZ```.
+
+Usage: ```basicDateTimeNoMillis($1)```
+
+Example: ```basicDateTimeNoMillis('20150101T000000Z')```
+
+#### dateHourMinuteSecondMillis
+
+Description: Formatter for full date, and time keeping the first 3 fractional seconds for format ```yyyy-MM-dd'T'HH:mm:ss.SSS```.
+
+Usage: ```dateHourMinuteSecondMillis($1)```
+
+Example: ```dateHourMinuteSecondMillis('2015-01-01T00:00:00.000')``` 
+
+#### millisToDate
+
+<<<<<<< HEAD
+Description: Create a new date from a long representing milliseconds since January 1, 1970.
+=======
+Description: Create a new date from as long representing millis since 1970.
+>>>>>>> master
```

Comment:
I thought that I fixed them.  It's just slight doc changes to avoid 
confusion with the new function secsToDate.

On 12/23/2015 05:42 PM, David Seapy wrote:
>
> In geomesa-convert/README.md~ 
> <https://github.com/locationtech/geomesa/pull/742#discussion_r48384907>:
>
> > +
> > +#### dateHourMinuteSecondMillis
> > +
> > +Description: Formatter for full date, and time keeping the first 3 fractional seconds for format ```yyyy-MM-dd'T'HH:mm:ss.SSS```.
> > +
> > +Usage: ```dateHourMinuteSecondMillis($1)```
> > +
> > +Example: ```dateHourMinuteSecondMillis('2015-01-01T00:00:00.000')```
> > +
> > +#### millisToDate
> > +
> > +<<<<<<< HEAD
> > +Description: Create a new date from a long representing milliseconds since January 1, 1970.
> > +=======
> > +Description: Create a new date from as long representing millis since 1970.
> > +>>>>>>> master
>
> there's a couple conflicts here and line 389 in Transformers.scala
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/locationtech/geomesa/pull/742/files#r48384907>.
>

-- 
Nicholas Elias, PhD

Commonwealth Computer Research, Inc. (CCRi)
1440 Sachem Place
Charlottesville, VA 22901

Phone: (434) 977 0600
E-mail: nelias@ccri.com